### PR TITLE
Bugfix/headers should be case insensitive

### PR DESCRIPTION
**Pull request by marabooy**
### Issues

*This pull request fixes issue #1882 .*

### Description

*Briefly describe the changes of this pull request.*

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

**Review by paulodero — CHANGES_REQUESTED** [review]
I noticed the rolling pipeline was cancelled before execution was complete. Could you run the build pipeline to the end to ensure all the tests are executed before opening this PR for review.

File: src/Microsoft.OData.Core/Batch/ODataBatchOperationHeaders.cs [hunk 1]
```diff
@@ -75,7 +71,7 @@ public void Add(string key, string value)
         /// <remarks>This method will only try to match the key using case-sensitive comparison.</remarks>
         public bool ContainsKeyOrdinal(string key)
```
Comment (paulodero) [inline src/Microsoft.OData.Core/Batch/ODataBatchOperationHeaders.cs:72]: The `xml` documentation of this method still indicate `...case-sensitive comparison..`.  Should be changed.

File: src/Microsoft.OData.Core/Batch/ODataBatchOperationHeaders.cs [hunk 2]
```diff
@@ -108,19 +98,13 @@ public bool Remove(string key)
...
-            key = this.FindKeyIgnoreCase(key);
-            if (key == null)
-            {
-                value = null;
-                return false;
-            }
-
-            return this.caseSensitiveDictionary.TryGetValue(key, out value);
+            value = null;
+            return false;
```
Comment (paulodero) [inline src/Microsoft.OData.Core/Batch/ODataBatchOperationHeaders.cs:107]: Why not just do `return this.headersDictionary.TryGetValue(key, out value);`

File: src/Microsoft.OData.Core/Batch/ODataBatchOperationHeaders.cs [hunk 3]
```diff
@@ -85,18 +81,12 @@ public bool ContainsKeyOrdinal(string key)
...
             }
 
-            key = this.FindKeyIgnoreCase(key);
-            if (key == null)
-            {
-                return false;
-            }
-
-            return this.caseSensitiveDictionary.Remove(key);
+            return false;
```
Comment (paulodero) [inline src/Microsoft.OData.Core/Batch/ODataBatchOperationHeaders.cs:89]: Could be simplified to `return this.headersDictionary.Remove(key);`

**Review by paulodero — CHANGES_REQUESTED** [review]

**Review by mikepizzo — APPROVED** [review]
:shipit:

**Comment by mikepizzo** [comment]
Build is passing and it looks like all comments have been addressed.  @paulodero please re-review and sign-off if you are satisfied with the resolutions.

File: src/Microsoft.OData.Core/Batch/ODataBatchOperationHeaders.cs [hunk 4]
```diff
@@ -64,18 +59,18 @@ public ODataBatchOperationHeaders()
...
         }
 
         /// <summary>
-        /// Determines whether the dictionary contains an element with the specified key using case-sensitive comparison.
+        /// Determines whether the dictionary contains an element with the specified key.
         /// </summary>
         /// <param name="key">The key to locate in the dictionary.</param>
         /// <returns>true if the dictionary contains an element with the <paramref name="key"/>; otherwise, false.</returns>
-        /// <remarks>This method will only try to match the key using case-sensitive comparison.</remarks>
+        /// <remarks>This method will only try to match the key using case-insensitive comparison.</remarks>
```
Comment (gathogojr) [inline src/Microsoft.OData.Core/Batch/ODataBatchOperationHeaders.cs:70]:
```suggestion
        /// <remarks>This method will match the key using case-insensitive comparison.</remarks>

```

**Review by gathogojr — APPROVED** [review]

**Review by paulodero — APPROVED** [review]
:shipit:

**Pull request merged**